database_manager/src/run_server.cpp

```diff
@@ -1775,11 +1771,7 @@ void update_world_states(ros::NodeHandle& node, std::vector<ToasterFactReader*>
         for (int i = 0; i < newState.size(); ++i) {
             if ((previousFactsState[j].subjectId == newState[i].subjectId)
                     && (previousFactsState[j].property == newState[i].property)
-                    && (previousFactsState[j].propertyType == newState[i].propertyType)
-                    && (previousFactsState[j].subProperty == newState[i].subProperty)
-                    && (previousFactsState[j].subjectOwnerId == newState[i].subjectOwnerId)
-                    && (previousFactsState[j].targetId == newState[i].targetId)
-                    && (previousFactsState[j].targetOwnerId == newState[i].targetOwnerId)) {
+                    && (previousFactsState[j].targetId == newState[i].targetId)) {
```

Comment:
Well, then it could also not work with targetId, it's not always set.
A vetter way would be to change the way we compare, maybe using string compare?[http://www.cplusplus.com/reference/string/string/compare/]